src/Sylius/Component/Core/Dashboard/SalesDataProvider.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Dashboard;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+
+final class SalesDataProvider implements SalesDataProviderInterface
+{
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    public function __construct(OrderRepositoryInterface $orderRepository)
+    {
+        $this->orderRepository = $orderRepository;
+    }
+
+    public function getLastYearSalesSummary(ChannelInterface $channel): SalesSummary
+    {
+        $result = $this->orderRepository->getLastYearSalesPerMonthForChannel($channel);
```

Comment:
What about if we would paste here the implementation of `getLastYearSalesPerMonthForChannel`? It's not connected that closely to the OrderRepository and that method does not have good enough contract (it returns some array) to rely on it here.

Reply:
Makes sense 👍